src/nupic/algorithms/Connections.cpp

```diff
@@ -103,7 +126,7 @@ Synapse Connections::createSynapse(Segment segment,
   SynapseData &synapseData = synapses_[synapse];
   synapseData.segment = segment;
   synapseData.presynapticCell = presynapticCell;
-  synapseData.permanence = permanence;
+  synapseData.permanence = minPermanence - 1; // Start in disconnected state.
```

Comment:
- if this is correct behavior, then permanence from arguments is not used, so remove (changes method signature -> API_Changelog) 
- also, permanence should always be within [min/maxPermanence], can you uses minPermanence as the default? (-> disconnected <=> minPermanence < connectedThreshold)

Reply:
...OK, permanence is eventually used: https://github.com/htm-community/nupic.cpp/pull/153/files/a428461a7458d60112d6acbd7943eb0911ec77ba..2ccdb9992738866337b4957ba71702fb486684aa#diff-dc7539503f4135f1c566ef5ba871fc7fR141

Reply:
Ok, maybe use just something more explicit? `const Permanence DISCONNECTED = Real32::MIN` ?

Reply:
How about: `connectedThreshold_ - 1`

Reply:
The purpose of this line is to ensure that the synapse starts as a disconnected synapse.  Setting the init permanence to less than the connectedThreshold guarantees that it starts disconnected.